xmtp_id/src/associations/association_log.rs

```diff
@@ -69,35 +65,34 @@ impl IdentityAction for CreateInbox {
             return Err(AssociationError::MultipleCreate);
         }
 
-        let account_address = self.account_address.clone();
-        let recovered_signer = self.initial_address_signature.recover_signer()?;
-        if recovered_signer.ne(&MemberIdentifier::Address(account_address.clone())) {
+        if self
+            .signature
+            .get_signer()
+            .ne(&MemberIdentifier::Address(self.account_address.clone()))
+        {
             return Err(AssociationError::MissingExistingMember);
         }
 
-        allowed_signature_for_kind(
-            &MemberKind::Address,
-            &self.initial_address_signature.signature_kind(),
-        )?;
+        allowed_signature_for_kind(&MemberKind::Address, &self.signature.get_kind())?;
 
-        if self.initial_address_signature.signature_kind() == SignatureKind::LegacyDelegated
-            && self.nonce != 0
-        {
+        if self.signature.get_kind() == &SignatureKind::LegacyDelegated && self.nonce != 0 {
             return Err(AssociationError::LegacySignatureReuse);
         }
 
-        Ok(AssociationState::new(account_address, self.nonce))
+        Ok(AssociationState::new(
+            self.account_address.clone(),
+            self.nonce,
+        ))
     }
 
     fn signatures(&self) -> Vec<Vec<u8>> {
-        vec![self.initial_address_signature.bytes()]
+        vec![self.signature.to_bytes()]
     }
 }
 
 /// AddAssociation Action
 pub struct AddAssociation {
     pub new_member_signature: Box<dyn Signature>,
-    pub new_member_identifier: MemberIdentifier,
```

Comment:
We need this to get the signature text, which we need to generate before we can verify the signature